packages/opentelemetry-plugin-document-load/src/documentLoad.ts

```diff
@@ -0,0 +1,283 @@
+          {
+            parent: connectSpan,
+          }
+        );
+      }
+    }
+    // Connection negotiation ends
+    this._endSpan(connectSpan, PerformanceTimingNames.CONNECT_END, entries);
+
+    // Cache seek plus response time
+    this._startAndFinishSpan(
+      AttributeNames.CACHE_SEEK,
+      PerformanceTimingNames.FETCH_START,
+      PerformanceTimingNames.RESPONSE_END,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    // TTFB - time to first byte
+    this._startAndFinishSpan(
+      AttributeNames.TTFB,
+      PerformanceTimingNames.REQUEST_START,
+      PerformanceTimingNames.RESPONSE_START,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    // Response time only (download)
+    this._startAndFinishSpan(
+      AttributeNames.RESPONSE_TIME,
+      PerformanceTimingNames.RESPONSE_START,
+      PerformanceTimingNames.RESPONSE_END,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    this._endSpan(rootSpan, PerformanceTimingNames.LOAD_EVENT_START, entries);
+  }
+
+  /**
+   * Helper function for ending span
+   * @param {Span} span
+   * @param {string} performanceName - name of performance entry for time end
+   * @param {PerformanceEntries} entries
+   * @private
+   */
+  private _endSpan(
+    span: Span | undefined,
+    performanceName: string,
+    entries: PerformanceEntries
+  ) {
+    if (typeof span !== 'undefined' && hasKey(entries, performanceName)) {
+      span.addEvent(performanceName);
+      span.end(entries[performanceName]);
+    }
+  }
+
+  /**
+   * gets performance entries of navigation
+   * @private
+   */
+  private _getEntries() {
+    const entries: PerformanceEntries = {};
+    const performanceNavigationTiming = otperformance.getEntriesByType(
+      'navigation'
+    )[0] as PerformanceEntries;
+    if (performanceNavigationTiming) {
+      const keys = Object.values(PerformanceTimingNames);
+      const startTime = otperformance.timeOrigin;
+      keys.forEach((key: string) => {
+        if (hasKey(performanceNavigationTiming, key)) {
+          const value = performanceNavigationTiming[key];
+          if (typeof value === 'number' && value > 0) {
+            entries[key] = value + startTime;
+          }
+        }
+      });
+    } else {
+      // // fallback to previous version
+      const perf: (typeof otperformance) & PerformanceLegacy = otperformance;
+      const performanceTiming = perf.timing;
+      if (performanceTiming) {
+        const keys = Object.values(PerformanceTimingNames);
+        keys.forEach((key: string) => {
+          if (hasKey(performanceTiming, key)) {
+            const value = performanceTiming[key];
+            if (typeof value === 'number' && value > 0) {
+              entries[key] = value;
+            }
+          }
+        });
+      }
+    }
+    return entries;
+  }
+
+  /**
+   * Helper function for starting and finishing a span
+   * @param {string} spanName - name of span
```

Comment:
```suggestion
   * @param spanName name of span
```

Reply:
This is to make things consistent everywhere. If you agree, could you pls make appropriate change in other places.

Reply:
hmm but why do you want to remove the type of param then ? All params here have such structure
`{type} name - description`
or if they don't have description
`{type} name`